### PR TITLE
docs: 4.0.0-rc.1 changelog

### PR DESCRIPTION
**Pull request by ycjcl868**
## 4.0.0-rc.1

`2020-01-11`

- 🌟 Drawer Added `footer` and `footerStyle` properties. [#20690](https://github.com/ant-design/ant-design/pull/20690) [@DeanVanNiekerk](https://github.com/DeanVanNiekerk)
- 🌟 Switch Added `@switch-min-width` and `@switch-sm-min-width` less variables. [#20829](https://github.com/ant-design/ant-design/pull/20829) [@abdih](https://github.com/abdih)
- Table
  - 🐞 Fix expanded icon not work when `expandRowByClick` is set. [#20808](https://github.com/ant-design/ant-design/pull/20808)
  - 🐞 Fix expanded row width in scaled dom element and border style. [#20805](https://github.com/ant-design/ant-design/pull/20805)
  - 🐞 Fix background color css priority too high to override user customize style. [#20794](https://github.com/ant-design/ant-design/pull/20794)
  - 🐞 Fix `rowSelection` of `fixed` not work. [#20735](https://github.com/ant-design/ant-design/pull/20735)
  - 🐞 Fix fixed columns in Chrome show the vertical scrollbar. [#20705](https://github.com/ant-design/ant-design/pull/20705)
  - 🐞 Fix crash when columns with empty children. [#20703](https://github.com/ant-design/ant-design/pull/20703)
- 💄 Tweak Calendar basic style month drop-down box width and notice items of word order and card mode, select the size of the box. [#20790](https://github.com/ant-design/ant-design/pull/20790) [@xrkffgg](https://github.com/xrkffgg)
- DatePicker
  - 💄 Optimize the `border-radius` of the rounded corners connection. [#20736](https://github.com/ant-design/ant-design/pull/20736)
  - 🐞 Fix selected style overlap. [#20736](https://github.com/ant-design/ant-design/pull/20736)
  - 🐞 Fix extra dividing line at the bottom. [#20736](https://github.com/ant-design/ant-design/pull/20736)
  - 🐞 Fix button style for DatePicker's default range. [#20760](https://github.com/ant-design/ant-design/pull/20760) [@xrkffgg](https://github.com/xrkffgg)
- 🐞 Input not block user input when value is `undefined`. [#20783](https://github.com/ant-design/ant-design/pull/20783)
- 🐞 Fix Carousel card carousel orientation in left / right mode. [#20781](https://github.com/ant-design/ant-design/pull/20781) [@xrkffgg](https://github.com/xrkffgg)
- 🐞 Fix Grid responsive gutter under SSR initial value of `0`. [#20762](https://github.com/ant-design/ant-design/pull/20762)
- 🐞 Fix InputNumber, Select and other components icon size. [#20765](https://github.com/ant-design/ant-design/pull/20765)
- 🐞 Fix Badge `z-index` higher than Table fixed columns. [#20751](https://github.com/ant-design/ant-design/pull/20751)
- 💄 Tweak font family to be same as tailwindcss. [#20747](https://github.com/ant-design/ant-design/pull/20747)
- 🐞 Fix TextArea `autoSize` blink in FireFox. [#20737](https://github.com/ant-design/ant-design/pull/20737)
- 🐞 Fix Form.Item not keep error message sync when hit rule changed. [#20725](https://github.com/ant-design/ant-design/pull/20725)
- 🐞 Fix Form.Item add additional feedback style when `hasFeedback` is not set. [#20691](https://github.com/ant-design/ant-design/pull/20691)
- 🐞 Fix Cascader search bug when `fieldNames` is existed and label/value share same name. [#20720](https://github.com/ant-design/ant-design/pull/20720)
- 🐞 Fix Collapse background color with wrong less variable. [#20718](https://github.com/ant-design/ant-design/pull/20718) [@kuitos](https://github.com/kuitos)
- 🐞 Fix Slider's Tooltip not follow handle. [#20699](https://github.com/ant-design/ant-design/pull/20699)
- 🐞 Fix Card cover image get skretched. [#20701](https://github.com/ant-design/ant-design/pull/20701)
- 🐞 Fix Typography automatic overflow with `suffix` property. [#20689](https://github.com/ant-design/ant-design/pull/20689) [@zouxiaomingya](https://github.com/zouxiaomingya)
- 🐞 Fix AutoComplete with customize component with wrong style. [#20686](https://github.com/ant-design/ant-design/pull/20686)
- 🐞 Fix Input.Group to be partial to a pixel in Form. [#20681](https://github.com/ant-design/ant-design/pull/20681)
- TypeScript
  - 🐞 Export Form interfaces. [3a1c5](https://github.com/ant-design/ant-design/commit/3a1c51010fecfa59f63f5e09027805a141e9ec11)
  - 🐞 Fix Table types. [#20789](https://github.com/ant-design/ant-design/pull/20789)
  - 🐞 Fix Table.Column and Table.ColumnGroup definition. [#20695](https://github.com/ant-design/ant-design/pull/20695)

### CODE REVIEW

File: CHANGELOG.en-US.md [hunk 1]
```diff
@@ -15,6 +15,45 @@ timeline: true
...
 
+## 4.0.0-rc.1
+
+`2020-01-11`
+
+- 🌟 Drawer Added `footer` and `footerStyle` properties. [#20690](https://github.com/ant-design/ant-design/pull/20690) [@DeanVanNiekerk](https://github.com/DeanVanNiekerk)
+- Table
+  - 🐞 Fix expanded icon not work when `expandRowByClick` is set. [#20808](https://github.com/ant-design/ant-design/pull/20808)
+  - 🐞 Fix expanded row width in scaled dom element and border style. [#20805](https://github.com/ant-design/ant-design/pull/20805)
+  - 🐞 Fix background color css priority to high to override user customize style. [#20794](https://github.com/ant-design/ant-design/pull/20794)
```
Comment (zombieJ) [inline CHANGELOG.en-US.md]: too

File: CHANGELOG.en-US.md [hunk 2]
```diff
@@ -15,6 +15,45 @@ timeline: true
...
+  - 🐞 Fix `rowSelection` of `fixed` not work. [#20735](https://github.com/ant-design/ant-design/pull/20735)
+  - 🐞 Fix fixed columns in Chrome show the vertical scrollbar. [#20705](https://github.com/ant-design/ant-design/pull/20705)
+  - 🐞 Fix crash when columns with empty children. [#20703](https://github.com/ant-design/ant-design/pull/20703)
+- 💄 Tweak Calendar basic style month drop-down box width and notice items of word order and card mode, select the size of the box。[#20790](https://github.com/ant-design/ant-design/pull/20790) [@xrkffgg](https://github.com/xrkffgg)
```
Comment (zombieJ) [inline CHANGELOG.en-US.md]: 。

File: CHANGELOG.en-US.md [hunk 3]
```diff
@@ -15,6 +15,45 @@ timeline: true
...
+- DatePicker
+  - 💄 Optimize the border-radius of the connection and change it to rounded corners. [#20736](https://github.com/ant-design/ant-design/pull/20736)
+  - 🐞 Fix selected style overlap. [#20736](https://github.com/ant-design/ant-design/pull/20736)
+  - 🐞 Fix extra dividing line at the bottom. [#20736](https://github.com/ant-design/ant-design/pull/20736)
+  - 🐞 Fix button style for DatePicker's default range. [#20760](https://github.com/ant-design/ant-design/pull/20760) [@xrkffgg](https://github.com/xrkffgg)
+- 🐞 Input not block user input when value is `undefined`. [#20783](https://github.com/ant-design/ant-design/pull/20783)
+- 🐞 Fix Carousel card carousel orientation in left / right mode. [#20781](https://github.com/ant-design/ant-design/pull/20781) [@xrkffgg](https://github.com/xrkffgg)
+- 🐞 Fix Grid responsive gutter under SSR initial value of 0. [#20762](https://github.com/ant-design/ant-design/pull/20762)
```
Comment (zombieJ) [inline CHANGELOG.en-US.md]: `0`

File: CHANGELOG.en-US.md [hunk 4]
```diff
@@ -15,6 +15,45 @@ timeline: true
...
+- 🐞 Fix InputNumber, Select and other components icon size. [#20765](https://github.com/ant-design/ant-design/pull/20765)
+- 🐞 Fix Badge zIndex higher than Table fixed columns. [#20751](https://github.com/ant-design/ant-design/pull/20751)
```
Comment (zombieJ) [inline CHANGELOG.en-US.md]: `z-index`

File: CHANGELOG.en-US.md [hunk 5]
```diff
@@ -15,6 +15,45 @@ timeline: true
...
+- 💄 Tweak font family to be same as tailwindcss. [#20747](https://github.com/ant-design/ant-design/pull/20747)
+- 🐞 Fix TextArea autoSize blink in FireFox. [#20737](https://github.com/ant-design/ant-design/pull/20737)
```
Comment (zombieJ) [inline CHANGELOG.en-US.md]: `autoSize`

File: CHANGELOG.en-US.md [hunk 6]
```diff
@@ -15,6 +15,45 @@ timeline: true
...
+- 🐞 Fix Form.Item not keep error message sync when hit rule changed. [#20725](https://github.com/ant-design/ant-design/pull/20725)
+- 🐞 Fix Form.Item add additional feedback style when `hasFeedback` is not set. [#20691](https://github.com/ant-design/ant-design/pull/20691)
+- 🐞 Fix Cascader search bug when `fieldNames` is existed and label/value share same name. [#20720](https://github.com/ant-design/ant-design/pull/20720)
+- 🐞 Fix Collapse backgroundColor Less variable. [#20718](https://github.com/ant-design/ant-design/pull/20718) [@kuitos](https://github.com/kuitos)
```
Comment (zombieJ) [inline CHANGELOG.en-US.md]: background color with wrong less variable

File: CHANGELOG.zh-CN.md [hunk 7]
```diff
@@ -15,6 +15,45 @@ timeline: true
...
+- Table
+  - 🐞 修复在 `expandRowByClick` 下展开 Icon 点击失效。[#20808](https://github.com/ant-design/ant-design/pull/20808)
+  - 🐞 修复在缩放下的展开行宽度样式。[#20805](https://github.com/ant-design/ant-design/pull/20805)
+  - 🐞 修复背景色优先级覆盖用户自定义背景样式。[#20794](https://github.com/ant-design/ant-design/pull/20794)
+  - 🐞 修复在 rowSelection 下 `fixed` 属性失效。[#20735](https://github.com/ant-design/ant-design/pull/20735)
+  - 🐞 修复固定列在 Chrome 下放大时出现纵向滚动条问题。[#20705](https://github.com/ant-design/ant-design/pull/20705)
+  - 🐞 修复 `columns` 为空时 Table 报错问题。[#20703](https://github.com/ant-design/ant-design/pull/20703)
+- 💄 优化 Calendar 基本样式月份下拉框宽度、通知事项的文字顺序以及卡片模式，选择框的尺寸。[#20790](https://github.com/ant-design/ant-design/pull/20790) [@xrkffgg](https://github.com/xrkffgg)
+- DatePicker
+  - 💄 优化连接处 border-radius，改成圆角。[#20736](https://github.com/ant-design/ant-design/pull/20736)
```
Comment (zombieJ) [inline CHANGELOG.zh-CN.md]: `border-radius`

File: CHANGELOG.zh-CN.md [hunk 8]
```diff
@@ -15,6 +15,45 @@ timeline: true
...
+## 4.0.0-rc.1
+
+`2020-01-11`
+
+- 🌟 Drawer 增加 `footer` 及 `footerStyle` 属性。[#20690](https://github.com/ant-design/ant-design/pull/20690) [@DeanVanNiekerk](https://github.com/DeanVanNiekerk)
+- Table
+  - 🐞 修复在 `expandRowByClick` 下展开 Icon 点击失效。[#20808](https://github.com/ant-design/ant-design/pull/20808)
+  - 🐞 修复在缩放下的展开行宽度样式。[#20805](https://github.com/ant-design/ant-design/pull/20805)
+  - 🐞 修复背景色优先级覆盖用户自定义背景样式。[#20794](https://github.com/ant-design/ant-design/pull/20794)
+  - 🐞 修复在 rowSelection 下 `fixed` 属性失效。[#20735](https://github.com/ant-design/ant-design/pull/20735)
```
Comment (zombieJ) [inline CHANGELOG.zh-CN.md]: `rowSelection`

File: CHANGELOG.en-US.md (hunk 1 above)
Comment (ycjcl868) [reply CHANGELOG.en-US.md]: ？

File: CHANGELOG.zh-CN.md [hunk 9]
```diff
@@ -15,6 +15,45 @@ timeline: true
...
+  - 🐞 修复在 `rowSelection` 下 `fixed` 属性失效。[#20735](https://github.com/ant-design/ant-design/pull/20735)
+  - 🐞 修复固定列在 Chrome 下放大时出现纵向滚动条问题。[#20705](https://github.com/ant-design/ant-design/pull/20705)
+  - 🐞 修复 `columns` 为空时 Table 报错问题。[#20703](https://github.com/ant-design/ant-design/pull/20703)
+- 💄 优化 Calendar 基本样式月份下拉框宽度、通知事项的文字顺序以及卡片模式，选择框的尺寸。[#20790](https://github.com/ant-design/ant-design/pull/20790) [@xrkffgg](https://github.com/xrkffgg)
+- DatePicker
+  - 💄 优化连接处 `border-radius`，改成圆角。[#20736](https://github.com/ant-design/ant-design/pull/20736)
+  - 🐞 修复选中样式重叠。[#20736](https://github.com/ant-design/ant-design/pull/20736)
+  - 🐞 修复底部额外分割线。[#20736](https://github.com/ant-design/ant-design/pull/20736)
+  - 🐞 修复预设范围的按钮样式。[#20760](https://github.com/ant-design/ant-design/pull/20760) [@xrkffgg](https://github.com/xrkffgg)
+- 🐞 修复 Input 值为 undefined 时不能输入的问题。[#20783](https://github.com/ant-design/ant-design/pull/20783)
```
Comment (afc163) [inline CHANGELOG.zh-CN.md]: undefined 加 `

File: CHANGELOG.zh-CN.md [hunk 10]
```diff
@@ -15,6 +15,45 @@ timeline: true
...
+- 🐞 修复 Carousel 组件 left/right 模式下卡片轮播方向。[#20781](https://github.com/ant-design/ant-design/pull/20781) [@xrkffgg](https://github.com/xrkffgg)
+- 🐞 修复 Grid 响应式 gutter 在 SSR 下初始值为 `0` zIndex h 的问题。[#20762](https://github.com/ant-design/ant-design/pull/20762)
```
Comment (afc163) [inline CHANGELOG.zh-CN.md]: zIndex h ?

File: CHANGELOG.zh-CN.md [hunk 11]
```diff
@@ -15,6 +15,45 @@ timeline: true
...
+- Table
+  - 🐞 修复在 `expandRowByClick` 下展开 Icon 点击失效。[#20808](https://github.com/ant-design/ant-design/pull/20808)
+  - 🐞 修复在缩放下的展开行宽度样式。[#20805](https://github.com/ant-design/ant-design/pull/20805)
+  - 🐞 修复背景色优先级高导致用户自定义样式被覆盖的问题。[#20794](https://github.com/ant-design/ant-design/pull/20794)
+  - 🐞 修复在 `rowSelection` 下 `fixed` 属性失效。[#20735](https://github.com/ant-design/ant-design/pull/20735)
+  - 🐞 修复固定列在 Chrome 下放大时出现纵向滚动条问题。[#20705](https://github.com/ant-design/ant-design/pull/20705)
+  - 🐞 修复 `columns` 为空时 Table 报错问题。[#20703](https://github.com/ant-design/ant-design/pull/20703)
+- 💄 优化 Calendar 基本样式月份下拉框宽度、通知事项的文字顺序以及卡片模式，选择框的尺寸。[#20790](https://github.com/ant-design/ant-design/pull/20790) [@xrkffgg](https://github.com/xrkffgg)
+- DatePicker
+  - 💄 优化连接处 `border-radius`，改成圆角。[#20736](https://github.com/ant-design/ant-design/pull/20736)
```
Comment (afc163) [inline CHANGELOG.zh-CN.md]: 『改成圆角』描述是不是错的？

**Pull request merged**